Alicerce/Sources/View/CollectionReusableView.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  CollectionReusableView.swift
+//  Alicerce
+//
+//  Created by Luís Afonso on 03/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import UIKit
+
+open class CollectionReusableView: UICollectionReusableView {
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        setupLayout()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+
+        setupLayout()
+    }
+}
+
+extension CollectionReusableView: View {
+    public func setupLayout() {
+        preconditionFailure("💥 you forgot to override the method? 😱")
```

Comment:
I would rephrase these messages to: `💥: Did you forget to override this method? 😱`

Also, I would perhaps change these to `assertionFailure`'s so that it won't crash in Release builds.

This should (hopefully) only crash in Debug builds, anyway, right? 😝